packages/grid/data-grid/src/DataGrid.tsx

```diff
@@ -56,6 +54,7 @@ const DataGrid2 = React.forwardRef<HTMLDivElement, DataGridProps>(function DataG
 });
 
 DataGrid2.propTypes = {
+  // @ts-expect-error to fix
```

Comment:
This should do it https://github.com/mui-org/material-ui/pull/23123.

Reply:
better to keep the require than silent the compiler.

Reply:
"silent the compiler" is concerning us, "keeping the require" breaks people's code #447. I would say the latter is more important.
https://github.com/mui-org/material-ui/pull/23123 will land this weekend. We can wait for its release if you prefer.

Reply:
I think that I have found a great tradeoff. TypeScript should leave the prop-type alone as the objective is to complement it. In https://github.com/mui-org/material-ui-x/pull/448/files#r509196367 we have seen that the expected behavior would be to have TypeScript errors. So, instead of having to handle the need to add `@ts-expect-error` we can make the whole block as an any, basically, like here https://github.com/mui-org/material-ui-pickers/blob/537dc6e8385dc805f37d9afa20aa3c6cd3898a7c/lib/src/wrappers/DesktopWrapper.tsx#L52-L55.